packages/mgt-components/README.md

```diff
@@ -55,6 +55,167 @@ The components can be used on their own, but they are at their best when they ar
+
+A complete example is available in the [React SharePoint Web Part Sample](https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/main/samples/sp-webpart/src/webparts/mgtDemo/MgtDemoWebPart.ts).
+
+```ts
+// [...] trimmed for brevity
+import { Providers } from '@microsoft/mgt-element/dist/es6/providers/Providers';
+import { customElementHelper } from '@microsoft/mgt-element/dist/es6/components/customElementHelper';
+import { SharePointProvider } from '@microsoft/mgt-sharepoint-provider/dist/es6/SharePointProvider';
+import { lazyLoadComponent } from '@microsoft/mgt-spfx-utils';
+
+// Async import of component that imports the React Components
+const MgtDemo = React.lazy(() => import('./components/MgtDemo'));
+
+export interface IMgtDemoWebPartProps {
+  description: string;
+}
+// set the disambiguation before initializing any webpart
+customElementHelper.withDisambiguation('bar');
+
+export default class MgtDemoWebPart extends BaseClientSideWebPart<IMgtDemoWebPartProps> {
+  // set the global provider
+  protected async onInit() {
+    if (!Providers.globalProvider) {
+      Providers.globalProvider = new SharePointProvider(this.context);
+    }
+  }
+
+  public render(): void {
+    const element = lazyLoadComponent(MgtDemo, { description: this.properties.description });
+
+    ReactDom.render(element, this.domElement);
+  }
+
+  // [...] trimmed for brevity
+}
+```
+
+### Dynamic imports aka Lazy Loading
+
+Using dynamic imports you can load dependencies asynchronously. This pattern allows you to load dependencies only when needed. For example, you may want to load a component only when a user clicks a button. This is a great way to reduce the initial load time of your application. In the context of disambiguation, you need to use this technique because components register themselves in the browser when they are imported.
+
+**Important:** If you import the components before you have applied the disambiguation, the disambiguation will not be applied and using the disambiguated tag name will not work.
+
+When using an `import` statement the import statement is hoisted and executed before any other code in the code block. To use dynamic imports you must use the `import()` function. The `import()` function returns a promise that resolves to the module. You can also use the `then` method to execute code after the module is loaded and the `catch` method to handle any errors if necessary.
+
+```typescript
+// static import via a statement
+import { Providers, customElementHelper } from '@microsoft/mgt-element';
+import { Msal2Provider } from '@microsoft/mgt-msal2-provider';
+
+customElementHelper.withDisambiguation('contoso');
+Providers.globalProvider = new Msal2Provider({clientId: 'clientId'});
+
+// dynamic import via a function
+import('@microsoft/mgt-components').then(() => {
+  // code to execute after the module is loaded
```

Comment:
I've assumed that you were meaning to add a separate static import example as we can't mix static and dynamic imports of `mgt-components` as it will break the application.

Reply:
@waldekmastykarz if you could let me know if this hits the mark or not that would be fantastic.

Reply:
Right, sorry for not being clearer. What I meant was to extend the sample with an example showing for example `<mgt-person/>` so that readers know where the contents of their app should go. You bring up a good point that it's not possible to mix both approach in a single code snippet.

Reply:
Thank you for the clarification, I think we're good on this one now